recirq/quantum_chess/measurement_utils.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2021 Google
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Util functions used for measurement.
+"""
+import cirq
+import numpy as np
+from itertools import product
+from scipy.linalg import kron
+from typing import Iterable
+
+def kron_product(Ms):
+    """Computes the Kronecker product of a given list of matrices.
+    """
+    if len(Ms) < 1:
+        raise ValueError("Expect a list of matrices to compute Kronecker product.")
+    result = Ms[0]
+    for M in Ms[1:]:
+        result = kron(result, M)
+    return result
+
+def pauli_decomposition(H, qs: Iterable[cirq.Qid]):
+    """Decompose the given measurement matrix into PauliStrings. 
+
+    Args:
+    H: 2-d matrix with row and column number equals to 2^n (n>0). 
+       E.g. H = np.array([[0.5+0.j,  0. +0.5j], [0. -0.5j,  0.5+0.j ]])
+    qs: a list of qubits with size == n.
+       E.g. qs = [a1]
+    """
+    if H.ndim != 2:
+        raise ValueError("Expect a 2-d matrix.")
```

Comment:
nit: Expect to Expected

It might be worth adding the function name for clarity also, but this is up to you:
`raise ValueError('pauli_decomposition expectx a 2-d matrix.')`

Reply:
Thanks. The function name is added to all error outputs.